File: app/assets/javascripts/OrganizationAffiliation.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const OrganizationAffiliationSchema = DataElementSchema({
  identifier: Any,
  active: Any,
  period: Any,
  organization: Any,
  participatingOrganization: Any,
  network: Any,
  code: Any,
  specialty: Any,
  location: Any,
  healthcareService: Any,
  telecom: Any,
  endpoint: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'OrganizationAffiliation' },

});

module.exports.OrganizationAffiliationSchema = OrganizationAffiliationSchema;
module.exports.OrganizationAffiliation = mongoose.model('OrganizationAffiliation', OrganizationAffiliationSchema);

File: app/models/qdm/organization_affiliation.rb
module QDM
  # app/models/qdm/organization_affiliation.rb
  class OrganizationAffiliation < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :identifier
    field :active
    field :period
    field :organization
    field :participatingOrganization
    field :network
    field :code
    field :specialty
    field :location
    field :healthcareService
    field :telecom
    field :endpoint
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
